src/app/plan/duck/sagas.ts

```diff
@@ -693,6 +719,274 @@ function* migrationPoll(action) {
+        }
+      } else {
+        updatedSpec.hooks = [newHook]
+      }
+
+      const updatedHooksSpec = {
+        spec: updatedSpec
+      }
+      return updatedHooksSpec
+    }
+
+    const patchPlanRes = yield client.patch(
+      new MigResource(MigResourceKind.MigPlan, migMeta.namespace),
+      currentPlan.metadata.name,
+      createHooksSpec()
+    );
+
+    yield put(PlanActions.addHookSuccess(createHookRes.data));
+    yield put(PlanActions.setCurrentPlan(patchPlanRes.data));
+    yield put(PlanActions.hookFetchRequest(patchPlanRes.data.spec.hooks));
+    yield put(AlertActions.alertSuccessTimeout('Successfully added a hook to plan.'));
+  } catch (err) {
+    yield put(PlanActions.addHookFailure(err));
+    yield put(AlertActions.alertErrorTimeout('Failed to add hook.'));
+  }
+}
+
+function* removeHookSaga(action) {
+  try {
+    const state = yield select();
+    const { migMeta } = state;
+    const { name } = action;
+    const client: IClusterClient = ClientFactory.cluster(state);
+
+    const migHookResource = new MigResource(MigResourceKind.MigHook, migMeta.namespace);
+
+    yield client.delete(
+      migHookResource,
+      name,
+    );
+
+    yield put(PlanActions.removeHookSuccess(name));
+    yield put(AlertActions.alertSuccessTimeout(`Successfully removed hook "${name}"!`));
+
+    const { currentPlan } = state.plan;
+
+    const createHooksSpec = () => {
+      const updatedSpec = Object.assign({}, currentPlan.spec);
+      if (updatedSpec.hooks) {
+        const deletedHookIndex = updatedSpec.hooks.findIndex((hook) => hook.reference.name === name)
+        updatedSpec.hooks.splice(deletedHookIndex, 1);
+      }
+
+      const updatedHooksSpec = {
+        spec: updatedSpec
+      }
+      return updatedHooksSpec
+    }
+
+    const patchPlanRes = yield client.patch(
+      new MigResource(MigResourceKind.MigPlan, migMeta.namespace),
+      currentPlan.metadata.name,
+      createHooksSpec()
+    );
+
+    yield put(PlanActions.removeHookSuccess(name));
+    yield put(PlanActions.setCurrentPlan(patchPlanRes.data));
+    yield put(PlanActions.hookFetchRequest(patchPlanRes.data.spec.hooks));
+
+
+  } catch (err) {
+    yield put(AlertActions.alertErrorTimeout(err));
+    yield put(PlanActions.removeHookFailure(err));
+  }
+}
+
+function* updateHookRequest(action) {
+  const state = yield select();
+  const { migMeta } = state;
+  const { migHook } = action;
+  const client: IClusterClient = ClientFactory.cluster(state);
+  const currentHook = state.plan.migHookList.find(hook => {
+    return hook.hookName === migHook.hookName;
+  });
+
+  const { currentPlan } = state.plan;
+
+  const currentPlanHookRef = currentPlan.spec.hooks.find(hook => {
+    return hook.reference.name === migHook.hookName;
+  })
+  const migHookObj = updateMigHook(
+    currentHook,
+    migHook,
+    currentPlanHookRef,
+    migMeta.namespace,
+    currentPlan
+  );
+  //need to patch entire hooks array 
```

Comment:
+1 this gets into patch strategy, there are different ways you can handle the array depending on it. I'm forgetting the lower level details, but there are a few ways you can tell k8s to handle it. I set the patch requester hardcoded to use `merge-patch+json`: https://github.com/konveyor/mig-ui/blob/master/src/client/client.ts#L43